### PR TITLE
Ft/modals/api keys

### PR DESCRIPTION
**Pull request by cwagner22**
### Modals/API Keys/CreateAPIKeysModal

image

image

### Modals/API Keys/EditAPIKeysModal

image

### CODE REVIEW

File: src/components/modals/APIKeysModal/CreateAPIKeysModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  }, []);
+
+  if (name) {
+    return (
+      <ModalContainer title="Save Your API Key" width="small" onClickClose={onClickClose}>
+        <Typography variant="body1" color="neutral200" weight="regular">
+          Save your secret in a safe place. After closing this window, you won’t have access your
+          secret anymore and will have to create a new API key.
+        </Typography>
+        <Gap gap={16}></Gap>
```
Comment (Xzirez) [inline src/components/modals/APIKeysModal/CreateAPIKeysModal/index.tsx:55]: close the attribute

File: src/components/modals/APIKeysModal/CreateAPIKeysModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    );
+  }
+
+  return (
+    <ModalContainer title="Create New API Key" width="small" onClickClose={onClickClose}>
+      <Typography variant="body1" color="neutral200" weight="regular">
+        Create a new API key to connect to your trading platform.
+      </Typography>
+      <Gap gap={16}></Gap>
+      <form noValidate onSubmit={handleSubmit(onFormSubmit)}>
```
Comment (Xzirez) [inline src/components/modals/APIKeysModal/CreateAPIKeysModal/index.tsx:88]: Im also wondering if we should use a standard form HTML element or a Form from the react-form package? I saw german used the react-form Form element.

File: src/components/modals/APIKeysModal/EditAPIKeysModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  } = methods;
+
+  const onFormSubmit = useCallback((data: any) => {
+    console.log(data);
+    onSubmit(data);
+  }, []);
+
+  return (
+    <ModalContainer title="Edit API Key Settings" width="small" onClickClose={onClickClose}>
+      <Gap gap={16}></Gap>
```
Comment (Xzirez) [inline src/components/modals/APIKeysModal/EditAPIKeysModal/index.tsx:42]: same here

File: src/components/modals/APIKeysModal/EditAPIKeysModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      <FormProvider {...methods}>
+        <form noValidate onSubmit={handleSubmit(onFormSubmit)}>
```
Comment (Xzirez) [inline src/components/modals/APIKeysModal/EditAPIKeysModal/index.tsx:44]: same here

File: src/components/modals/APIKeysModal/EditAPIKeysModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    reValidateMode: "onChange",
+  });
+  const {
+    handleSubmit,
+    formState: { isValid },
+    register,
+  } = methods;
+
+  const onFormSubmit = useCallback((data: any) => {
+    console.log(data);
```
Comment (Xzirez) [inline src/components/modals/APIKeysModal/EditAPIKeysModal/index.tsx:36]: remove. You can put it in the story, so we can test it in the future.

**Comment by Jennimar17** [comment]
image

**Comment by Jennimar17** [comment]
image

**Pull request merged**